### PR TITLE
WIP: add tests for client_may_init and client_may_update

### PR DESCRIPTION
**Pull request by jwshea**
This pull request is for feedback, there's probably a better way than the naive cmp_deeply I've used...

### CODE REVIEW

File: t/basic.t [hunk 1]
```diff
@@ -2097,4 +2097,110 @@ subtest "string normalization" => sub {
...
+    [ setBiscuits => {
+        create => {
+           foo => { type => 'anzac' },
+           bar => { type => 'anzac', qty => jnum(1) },
+        },
+    } ],
+  ]);
+
+  cmp_deeply(
+    $jmap_tester->strip_json_types( $res->as_pairs ),
```
Comment (rjbs) [inline t/basic.t]:
```
my $set = $res->single_sentence('biscuitsSet')->as_set;

my $foo_id =  $set->created_id('foo');
ok($foo_id, "we created foo");
jcmp_deeply(
  $set->create_errors->{bar},
  superhashof({
     propertyErrors => { qty => ignore() },
   }),
  }),
);
```

Comment (rjbs) [reply t/basic.t]: …and if it didn't work I might say:

```
unless ($tests_passed_or_whatever) {
  diag( explain( $res->as_stripped_pairs ));
}
```

**Comment by jwshea** [comment]
I'd like to squash these before merging, just keeping separate to ease review; wasn't sure if github pull requests handle history rewrites as nicely as gitlab.

**Comment by rjbs** [comment]
Yup, if you squash and force-push, it will track properly, at least in my conception of properly.

**Comment by rjbs** [comment]
I have squashed and applied.

**Pull request closed without merging**